cpu/cc2538/dev/uart.c

```diff
@@ -136,7 +136,9 @@
  * Baud rate defines used in uart_init() to set the values of UART_IBRD and
  * UART_FBRD in order to achieve the configured baud rates.
  */
-#define UART_CLOCK_RATE       16000000 /* 16 MHz */
+
+#define UART_CLOCK_RATE			(SYS_CTRL_32MHZ/(1 << SYS_CTRL_SYS_DIV))
+
```

Comment:
@g-oikonomou: No need to test. The answer is in table 7-2 and §7.3.1 in the UG. {UART_CC|SSI_CC}.CS.{DSEN|PIOSC} select the UART and SSI system and core clocks in the various modes. The UART driver uses the system clock, but the SPI driver uses the I/O clock. It would be better to also make the UART driver use the I/O clock.